homeassistant/components/deluge/sensor.py

```diff
@@ -27,25 +31,60 @@
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType, StateType
 
 from . import DelugeEntity
-from .const import DEFAULT_NAME, DEFAULT_RPC_PORT, DOMAIN
+from .const import (
+    CURRENT_STATUS,
+    DATA_KEYS,
+    DEFAULT_NAME,
+    DEFAULT_RPC_PORT,
+    DOMAIN,
+    DOWNLOAD_SPEED,
+    UPLOAD_SPEED,
+)
 from .coordinator import DelugeDataUpdateCoordinator
 
-SENSOR_TYPES: tuple[SensorEntityDescription, ...] = (
-    SensorEntityDescription(
-        key="current_status",
+
+def get_state(data: dict[str, float], key: str) -> str | float:
+    """Get current download/upload state."""
+    upload = data[DATA_KEYS[0]] - data[DATA_KEYS[2]]
+    download = data[DATA_KEYS[1]] - data[DATA_KEYS[3]]
+    if key == CURRENT_STATUS:
+        if upload > 0 and download > 0:
+            return "Up/Down"
+        if upload > 0 and download == 0:
+            return "Seeding"
+        if upload == 0 and download > 0:
+            return "Downloading"
```

Comment:
Side note: We don't allow calculating state in device or service integrating integrations, so this sensor type isn't really in line with standards.